### ISSUE TITLE
Html comments are shown in PR and issue descriptions

### ISSUE THREAD

**Issue by viceice** [go-gitea/gitea#19195]
### Description

Since v1.16.5 Html comments are shown in PR and issue descriptions, they should be hidden as before.

Reproducer: https://try.gitea.io/viceice/test/issues/1

### Gitea Version

1.16.5

### Can you reproduce the bug on the Gitea demo site?

Yes

### Screenshots

![image](https://user-images.githubusercontent.com/1798109/159889314-87fa78bd-368e-4743-83e6-ea13455642cd.png)

### How are you running Gitea?

Official helm chart with slim image

### Database

PostgreSQL

**Comment by viceice** [comment]
It seems `<` is encoded as `\x3C` 🤔

**Comment by xoxys** [comment]
As discussed in the chat:

```
gitea on  HEAD (e6d46ee) (BISECTING) [$] via 🐹 v1.16.15 via  v16.14.0 took 8s 
❯ git bisect good
f9ea4ab69ae397be4791c2e2ebf59f6cd9e5a6f8 is the first bad commit
commit f9ea4ab69ae397be4791c2e2ebf59f6cd9e5a6f8
Author: techknowlogick <techknowlogick@gitea.io>
Date:   Sat Mar 19 13:46:47 2022 -0400
```

**Comment by viceice** [comment]
maybe the `github.com/yuin/goldmark` v1.4.0 to v1.4.8 update?

**Comment by viceice** [comment]
I think this is fixed in a later version
- https://github.com/yuin/goldmark/issues/285
- https://github.com/yuin/goldmark/commit/e64a68fc130e0c74dd2cbec59e6737a2e2bb7398

**Comment by xoxys** [comment]
Should be a quick fix in this case.

**Comment by viceice** [comment]
When can i expect a fixed gitea version? If it needs some days i would create a small jquery script on page load to fix it temporary

**Comment by xoxys** [comment]
Don't know, one of the maintainers could give us a rough timeline.

**Comment by viceice** [comment]
Just ask if it's worth to creaste the quickfix 🙃

**Comment by xoxys** [comment]
Can confirm that updating goldmark fixes the issue:

```
❯ docker run -v $(pwd):/drone/src --workdir /drone/src -it golang:1.18 bash
root@c5219c2fabbe:/drone/src# go get -u github.com/yuin/goldmark
go: downloading github.com/yuin/goldmark v1.4.11
go: upgraded github.com/yuin/goldmark v1.4.8 => v1.4.11
root@c5219c2fabbe:/drone/src# TAGS="bindata sqlite sqlite_unlock_notify" make build
```

So you could build it with the fixed version from the `release/v1.16` branch on your own.

**Comment by viceice** [comment]
OK, thanks. Will do a jQuery quickfix for now. It's easier, as i already have a custom js file embedded. 😉 

```js
$('.task-list-item').contents().each((i, e) => {
  if (e.nodeName === '#text') {
    e.textContent = e.textContent.replaceAll(/\x3C!--.*?-->/g, '')
  }
})
```
